src/query/functions/temporal/base.go

```diff
@@ -350,13 +350,15 @@ func getIndices(
 	}
 
 	if r == -1 {
-		r = len(dp)
+		r = len(dps)
 	} else {
 		r = r + init
 	}
 
 	if leftBound {
 		l = l + init
+	} else {
+		return l, r, false
```

Comment:
What's this case covering?  Do you mind adding a comment here?

```
if leftBound { ...
} else {
  // For <reason> we just don't match if no left bound found?
  return ...
}
```

Reply:
Aight added in: `// if left bound was not found, there are no valid candidate points here.`